src/compiler/program.ts

```diff
@@ -1155,11 +1155,11 @@ namespace ts {
             forEach(rootNames, name => processRootFile(name, /*isDefaultLib*/ false));
 
             // load type declarations specified via 'types' argument or implicitly from types/ and node_modules/@types folders
-            const typeReferences: string[] = getAutomaticTypeDirectiveNames(options, rootNames, host);
+            const typeReferences: string[] = getAutomaticTypeDirectiveNames(options, host);
 
             if (typeReferences) {
-                const inferredRoot = getInferredTypesRoot(options, rootNames, host);
-                const containingFilename = combinePaths(inferredRoot, "__inferred type names__.ts");
+                // This containgFilename needs to matched with the one used in managed-side
+                const containingFilename = "__inferred type names__.ts"

```

Comment:
From talking off line, this is incorrect as it will not be unique to the project of interest. IT will be problematic if more than one project is open. The fix is to use `host.getCurrentDirectory()`